crates/rattler_solve/src/libsolv/wrapper/pool.rs

```diff
@@ -316,28 +236,49 @@ impl From<MatchSpecId> for Id {
     }
 }
 
+fn match_spec_to_c_string(match_spec: &MatchSpec) -> CString {
```

Comment:
Maybe we can move the conversion from MatchSpec to string into `Display` of `MatchSpec`. Im sure it will be used more.